modules/cudaarithm/CMakeLists.txt

```diff
@@ -9,6 +9,9 @@ ocv_warnings_disable(CMAKE_CXX_FLAGS /wd4127 /wd4324 /wd4512 -Wundef -Wmissing-d
 set(extra_dependencies "")
 set(optional_dependencies "")
 if(ENABLE_CUDA_FIRST_CLASS_LANGUAGE)
+  if(UNIX AND NOT BUILD_SHARED_LIBS AND CUDA_VERSION_STRING VERSION_GREATER_EQUAL 9.2 AND CMAKE_VERSION VERSION_GREATER_EQUAL 3.23)
```

Comment:
`CUDA_VERSION_STRING` is set as part of [FindCUDA ](https://cmake.org/cmake/help/latest/module/FindCUDA.html#result-variables).  When CUDA is a first class language and you are calling [FindCUDAToolkit](https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html#findcudatoolkit) you need to check the [`CUDAToolkit_VERSION...`](https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html#result-variables)

Reply:
it is set here https://github.com/opencv/opencv/blob/bef5a8768004c3d9b5eeb742446afe7d4e3e8145/cmake/OpenCVDetectCUDALanguage.cmake#L34
so it should work.

Reply:
Fair enough that can be refactored out in the future when *FindCuda* is completely removed.